app/components/challenge.js

```diff
@@ -87,7 +87,9 @@ export default class Challenge extends React.Component {
       }
 
       this.logActivity(results);
-    });
+    }
+    
+    Evaluator.run(this.state.src).then(handleResults, handleResults);
```

Comment:
Can we do `Evaluator.run(this.state.src).then(handleResults).catch(handleResults)` ?